zookeeper-recipes/zookeeper-recipes-queue/src/test/java/org/apache/zookeeper/recipes/queue/DistributedQueueTest.java

```diff
@@ -27,243 +28,260 @@
-
-        try {
-            queueHandles[0].remove();
-        } catch (NoSuchElementException e) {
-            return;
-        }
-
-        Assert.fail();
+    try {
+      queueHandles[0].remove();
+    } catch (NoSuchElementException e) {
+      return;
+    }
+    Assert.assertTrue(false);
+  }
+
+  public void createNremoveMtest(String dir, int n, int m) throws Exception {
+    String testString = "Hello World";
+    final int numClients = 2;
+    ZooKeeper[] clients = new ZooKeeper[numClients];
+    DistributedQueue[] queueHandles = new DistributedQueue[numClients];
+    for (int i = 0; i < clients.length; i++) {
+      clients[i] = createClient();
+      queueHandles[i] = new DistributedQueue(clients[i], dir, null);
     }
 
-    public void createNremoveMtest(String dir, int n, int m) throws Exception {
-        String testString = "Hello World";
-        final int numClients = 2;
-        ZooKeeper[] clients = new ZooKeeper[numClients];
-        DistributedQueue[] queueHandles = new DistributedQueue[numClients];
-        for (int i = 0; i < clients.length; i++) {
-            clients[i] = createClient();
-            queueHandles[i] = new DistributedQueue(clients[i], dir, null);
-        }
-
-        for (int i = 0; i < n; i++) {
-            String offerString = testString + i;
-            queueHandles[0].offer(offerString.getBytes());
-        }
+    for (int i = 0; i < n; i++) {
+      String offerString = testString + i;
+      queueHandles[0].offer(offerString.getBytes(UTF_8));
+    }
 
-        byte[] data = null;
-        for (int i = 0; i < m; i++) {
-            data = queueHandles[1].remove();
-        }
+    byte[] data = null;
+    for (int i = 0; i < m; i++) {
+      data = queueHandles[1].remove();
+    }
+    Assert.assertEquals(new String(data, UTF_8), testString + (m - 1));
+  }
+
+  @Test
+  public void testRemove2() throws Exception {
+    createNremoveMtest("/testRemove2", 10, 2);
+  }
+
+  @Test
+  public void testRemove3() throws Exception {
+    createNremoveMtest("/testRemove3", 1000, 1000);
+  }
+
+  public void createNremoveMelementTest(String dir, int n, int m)
+      throws Exception {
+    String testString = "Hello World";
+    final int numClients = 2;
+    ZooKeeper[] clients = new ZooKeeper[numClients];
+    DistributedQueue[] queueHandles = new DistributedQueue[numClients];
+    for (int i = 0; i < clients.length; i++) {
+      clients[i] = createClient();
+      queueHandles[i] = new DistributedQueue(clients[i], dir, null);
+    }
 
-        Assert.assertNotNull(data);
-        Assert.assertEquals(new String(data), testString + (m - 1));
+    for (int i = 0; i < n; i++) {
+      String offerString = testString + i;
+      queueHandles[0].offer(offerString.getBytes(UTF_8));
     }
 
-    @Test
-    public void testRemove2() throws Exception {
-        createNremoveMtest("/testRemove2", 10, 2);
+    byte[] data = null;
+    for (int i = 0; i < m; i++) {
+      data = queueHandles[1].remove();
```

Comment:
I think this creates an unused variable. You could remove the assignment, as it was before, or you can do something with it in a new Assert statement. A simple thing is just assert that it's not null.